core/src/main/java/org/apache/metamodel/query/FromItem.java

```diff
@@ -267,6 +267,11 @@ private void appendJoinOnItem(StringBuilder sb, String sideAlias, SelectItem onI
             sb.append(superQueryAlias);
             return;
         }
+        
+        if(_join != null && _leftSide.getJoin() != null) {
```

Comment:
If we do it like this, then I think we should have the same treatment also on the right side as we have on the left side.